libtenzir/src/plugin.cpp

```diff
@@ -548,12 +542,40 @@ class store_printer final : public plugin_printer {
   const store_plugin* plugin_;
 };
 
+auto store_plugin::parse_parser(parser_interface& p) const
+  -> std::unique_ptr<plugin_parser> {
+  argument_parser{name()}.parse(p);
+  return std::make_unique<store_parser>(this);
+}
+
 auto store_plugin::parse_printer(parser_interface& p) const
   -> std::unique_ptr<plugin_printer> {
   argument_parser{name()}.parse(p);
   return std::make_unique<store_printer>(this);
 }
 
+auto store_plugin::serialize(inspector& f, const plugin_printer& x) const
+  -> bool {
+  auto o = detail::overload{[&](auto g) {
+    return g.get().object(x).fields();
+  }};
+  auto obj = std::visit(o, f);
+  return f.apply(obj);
+}
+
+auto store_plugin::deserialize(inspector& f,
+                               std::unique_ptr<plugin_printer>& x) const
+  -> void {
+  auto o = detail::overload{[&](auto g) {
+    return g.get().object(x).fields();
+  }};
+  auto obj = std::visit(o, f);
+  if (f.apply(obj)) {
+    x = std::make_unique<store_printer>(this);
+  } else {
+    x = nullptr;
+  }
+}
```

Comment:
Much better! 

Nit: In the first call to `object(x)`, `x` is a `plugin_printer`, while in the second call, it's a `std::unique_ptr<plugin_printer>`. I don't think it matters for the inspectors that we are using, but in general, it would be best if both were the same type, perhaps even `store_printer` (deserialization then needs `x = std::make_unique` before inspecting `*x`, but that way the types are as accurate as possible).